### PR TITLE
Update display name on Android contact even if the field is empty

### PR DESCRIPTION
**Pull request by kosssi**

### CODE REVIEW

File: src/app/replicator/transformer/cozy_to_android_contact.coffee [hunk 1]
```diff
@@ -3,159 +3,243 @@ log = require('../../lib/persistent_log')
...
-    _n2ContactName: (n) ->
-        return undefined unless n?
+    # Cf cozy-vCard.nToFN :
+    validParts = parts.filter (part) -> part? and part isnt ''
+    formatted = validParts.join ' '
 
-        parts =  n.split ';'
-        [familyName, givenName, middle, prefix, suffix] = parts
+    contactName = new ContactName formatted, familyName, givenName, \
+        middle, prefix, suffix
```
Comment (nono) [inline src/app/replicator/transformer/cozy_to_android_contact.coffee]: Maybe

```coffee
contactName = new ContactName formatted, parts...
```

File: src/app/replicator/transformer/cozy_to_android_contact.coffee [hunk 2]
```diff
@@ -3,159 +3,243 @@ log = require('../../lib/persistent_log')
...
+        log.warning 'adr datapoint has bad type'
+        return new ContactAddress undefined, datapoint.type, ''
+
+
+# loop trought the cozy's datapoints list and fill up the cordovaContact
+# with
+_dataPoints2Cordova = (cozyContact, cordovaContact) ->
+    addContactField = (cordovaField, datapoint) ->
+        unless cordovaContact[cordovaField]
+            cordovaContact[cordovaField] = []
```
Comment (nono) [inline src/app/replicator/transformer/cozy_to_android_contact.coffee]:
```coffee
cordovaContact[cordovaField] ?= []
```

File: src/app/replicator/transformer/cozy_to_android_contact.coffee [hunk 3]
```diff
@@ -3,159 +3,243 @@ log = require('../../lib/persistent_log')
...
+        name = datapoint.name.toUpperCase()
+        switch name
+            when 'TEL'
+                addContactField 'phoneNumbers', datapoint
+
+            when 'EMAIL'
+                addContactField 'emails', datapoint
+            when 'ADR'
+                unless cordovaContact.addresses
+                    cordovaContact.addresses = []
```
Comment (nono) [inline src/app/replicator/transformer/cozy_to_android_contact.coffee]: Same here

File: src/app/replicator/transformer/cozy_to_android_contact.coffee (hunk 1 above)
Comment (kosssi) [inline src/app/replicator/transformer/cozy_to_android_contact.coffee]: I never see that :+1:

**Comment by kosssi** [comment]
thanks

**Pull request merged**